### ISSUE TITLE
Cannot convert JSON schema when using "$ref" syntax

### ISSUE THREAD

**Issue by HavenLin** [lifeomic/json-schema-to-graphql-types#21]
I have got error as "{ UnknownTypeReference: The referenced type #/definitions/metrics (undefined) is unknown in http://*/_Response.json.metrics".

does this tool support "$ref" syntax?

the schema example as :

    {
        "$schema": "http://json-schema.org/draft-04/schema#",
        "description": "A json-schema for ***** Response.",
        "id": "http://**/_Response.json",
        "definitions": {
            "metrics": {
                "type": "array",
                "items": {
                    "properties": {
                        "name": {
                            "type": "string"
                        },
                        "type": {
                            "enum": [
                                "A",
                                "B"
                            ],
                            "type": "string"
                        }
                    },
                    "required": [
                        "type",
                        "name"
                    ],
                    "type": "object"
                },
                "default": null
            }
        },
        "type": "object",
        "properties": {
            "metrics": {
                "$ref": "#/definitions/metrics"
            },
            "success": {
                "type": "boolean"
            }
        }
    }

**Comment by HavenLin** [comment]
@mdlavin I have created, any question/suggestion please feel free to let me know.

**Comment by MorganHuegel** [comment]
@mdlavin  I'm able to duplicate this error if I global install and run the command `convert-json-schemas-to-graphql-types <directory>` .  However, I am _not_ able to duplicate this error when I clone the repo and run the actual file `bin/convert-json-schemas-to-graphql-types.js <directory>` Output: 
```
type DefinitionMetricsItem {
  type: DefinitionMetricsItemType!
  name: String!
}

input DefinitionMetricsItemIn {
  type: DefinitionMetricsItemType!
  name: String!
}

enum DefinitionMetricsItemType {
  A
  B
}

type Http://**/ResponseJson {
  success: Boolean
  metrics: [DefinitionMetricsItem!]
}

input Http://**/ResponseJsonIn {
  success: Boolean
  metrics: [DefinitionMetricsItemIn!]
}

type Mutation {
  DefinitionMetricsIn(input: [DefinitionMetricsItemIn!]): String
  http://**/_Response.json(input: Http://**/ResponseJsonIn): String
}
```
I checked the package.json file, and the `bin` does seem to map the command to the correct file.  Could there be a discrepancy between the npm package and the up-to-date repo?

**Comment by mdlavin** [comment]
It's very possible that I either failed to push a new version or something
about the new version build failed. I'll release a new version today

**Comment by HavenLin** [comment]
actually, I want to output graphQL Query schema from provided example.

**Comment by mdlavin** [comment]
@MorganHuegel while I work on releasing a new version .. that outputed schema will definitely not be valid. Would you consider adding more validation on the names. You can check the GraphQL schema for details about what characters are valid in names

**Comment by MorganHuegel** [comment]
I looked into it, and the GraphQL type name must match this regex `/^[_a-zA-Z][_a-zA-Z0-9]*$/`, so basically either a number, letter, or `_`.  However, it mentions in the json-schema docs that best practice is to set the `$id` to an absolute URI ending in _.json_, which is what the users tend to do, but that won't match the regex.  Should we just try to split the file name from the _.json_ file, and throw out the rest of the URI?

**Comment by mdlavin** [comment]
I love the idea of extracting the file name, without trailing .json, if the
$id attribute is a URI. If the $id attribute is a valid GraphQL name, it
could be used as is. Really good suggestion!

**Comment by mdlavin** [comment]
@HavenLin I've released v1.2 of the library to help convert your schema. Can you try again? It's possible that your type name will not be valid, but @MorganHuegel has added code to provide you with better error messages if that's your problem
